scalableminds/webknossos
# Further Time Tacking Improvement

This PR adds the following suggest improvements to the time tracking view:
- [x] The format of the duration in the tooltip is now minutes:seconds
- [x] A spinning wheel is showing up when the data is being fetched from the backend and rendered into a diagram
- [x] There seems to be a bug that causes tooltips to not being rendered. Look at the [discuss](https://discuss.webknossos.org/t/a-few-updates-for-the-time-tracking-tool/1374/8?u=potto) for further clarification  

### URL of deployed dev instance (used for testing):
- [https://improvetimetracking.webknossos.xyz](https://improvetimetracking.webknossos.xyz)

### Steps to test:
- Get some tracking data by e.g. tracing.
- open the time tracking view and select the correct user and date.
- When the data is fetched by the server, there should be a spinning wheel visible. But it is not visible, while the google charts library is actually rendering the chart.
- The tooltip should always be next to the mouse (if visible). It should **not** appear somewhere else and then "beam" next to the mouse.
- The duration displayed in the tooltip should have the format of `minutes:seconds`.
- Also please consider that the tooltip might be partly positioned out of the window. I added a fix that adjusts the position for these cases.  Please also try to whether you can find a case where the tooltip is still partly placed outside the window.

### Issues:
- fixes #4076

------
